discovery/gossiper.go

```diff
@@ -3351,3 +3185,220 @@ func (d *AuthenticatedGossiper) validateChannelUpdate(nMsg *networkMsg,
 
 	return nil
 }
+
+// getAndValidateEdgeInfo queries the db for a channel edge info and its
+// corresponding policies. When the info is found and this is a remote message,
+// it will validate the announcement and check for rate limiting. When it's not
+// found, the message is put into `prematureChannelUpdates` and will be
+// reprocessed later when the corresponding channel announcement is received.
+func (d *AuthenticatedGossiper) getAndValidateEdgeInfo(nMsg *networkMsg,
+	upd *lnwire.ChannelUpdate, shortChanID uint64, ts time.Time,
+	graphScid lnwire.ShortChannelID) (*channeldb.ChannelEdgeInfo, error) {
+
+	chanInfo, e1, e2, err := d.cfg.Router.GetChannelByID(graphScid)
+	if err == nil {
+		// Exit early if this is a local channel update.
+		if !nMsg.isRemote {
+			return chanInfo, nil
```

Comment:
Why should we _not_ validate our own local announcements? This can potentially help us catch bugs introduced in say the funding manager, that actually violate the spec mandated verification that we perform here.

Reply:
Can see why we'd want to skip the rate limit below though however.

Reply:
> This can potentially help us catch bugs introduced in say the funding manager, that actually violate the spec mandated verification that we perform here.

Good point. Overlooked this part and now it's added back!